src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,398 @@
+                            c => ApplyFixAsync(context.Document, root, targetNode, convType)));
+                    }
+
+                    if (potentialConversionTypes.Length > MaximumConversionOptions)
+                    {
+                        // If the number of potential conversion types is larger than options we could show, report telemetry
+                        Logger.Log(FunctionId.CodeFixes_AddExplicitCast,
+                            KeyValueLogMessage.Create(m =>
+                            {
+                                m["NumberOfCandidates"] = potentialConversionTypes.Length;
+                            }));
+                    }
+
+                    context.RegisterCodeFix(new CodeAction.CodeActionWithNestedActions(
+                        CSharpFeaturesResources.Add_explicit_cast,
+                        actions.ToImmutableAndFree(), isInlinable: false),
+                        context.Diagnostics);
+                }
+            }
+        }
+
+        private static Task<Document> ApplyFixAsync(Document document, SyntaxNode currentRoot, ExpressionSyntax targetNode, ITypeSymbol conversionType)
+        {
+            var castExpression = targetNode.Cast(conversionType);
+            var newRoot = currentRoot.ReplaceNode(targetNode, castExpression.WithAdditionalAnnotations(Simplifier.Annotation));
+            return Task.FromResult(document.WithSyntaxRoot(newRoot));
+        }
+
+        /// <summary>
+        /// Output the current type information of the target node and the conversion type(s) that the target node is going to be cast by.
+        /// Implicit downcast can appear on Variable Declaration, Return Statement, and Function Invocation
+        /// </summary>
+        /// For example:
+        /// Base b; Derived d = [||]b;       
+        /// "b" is the current node with type "Base", and the potential conversion types list which "b" can be cast by is {Derived}
+        /// 
+        /// root: The root of the tree of nodes.
+        /// targetNode: The node to be cast.
+        /// targetNodeType: Output the type of "targetNode".
+        /// potentialConversionTypes: Output the potential conversions types that "targetNode" can be cast to
+        /// <returns>
+        /// True, if the target node has at least one potential conversion type, and they are assigned to "potentialConversionTypes"
+        /// False, if the target node has no conversion type.
+        /// </returns>
+        private static bool TryGetTargetTypeInfo(SemanticModel semanticModel, SyntaxNode? targetNode, CancellationToken cancellationToken,
+            out ITypeSymbol? targetNodeType, out ImmutableArray<ITypeSymbol> potentialConversionTypes)
+        {
+            targetNodeType = null;
+            potentialConversionTypes = ImmutableArray<ITypeSymbol>.Empty;
+            if (targetNode == null)
```

Comment:
Can `targetNode` be null?

Reply:
I just checked, and `targetNode` cannot be null, I am removing it